lib/exercism/user.rb

```diff
@@ -52,21 +52,21 @@ def access?(problem)
   end
 
   # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
-  def self.from_github(id, username, email, avatar_url, joined_as=nil)
-    user = User.where(github_id: id).first
+  def self.from_authentication_data(github_user, joined_as=nil)
```

Comment:
The rename came from another PR https://github.com/exercism/exercism.io/pull/3443

I a checklist part of this PR 
> Move logic of User model for authenticating with Github info to separate class, this method looks for some refactor.

Also, I don't know if it should be part of this PR. what do you think @Insti I'll be happy to do it of course